### ISSUE TITLE
Sending forms with BSN-prefill in read-only fields for birthday and postcode fails with 'Call did not validate on the backend'

### ISSUE THREAD

**Issue by hlmr-erik** [open-formulieren/open-forms#1899]
### Product versie / Product version

1.1.5

### Omschrijf het probleem / Describe the bug

When trying to send a form where the data has been pre-filled w/that of a BSN test account, sending the form fails with a notice 'Call did not validate on the backend'. Taking a look in the dev-tools, shows that it judges the fields for postcode and birthdate to be incorrect: **invalidPrefilledFields":["Geboortedatum","Postcode"]**

The fields don't show validation-errors in the form itself:
![image](https://user-images.githubusercontent.com/111565013/185928847-9aaee2c4-d13f-4e27-a42c-4fae4fca114b.png)

**Edit 14-09: The problem disappears when disabling read-only on these fields**_

### Stappen om te reproduceren / Steps to reproduce

1. Open the form https://haarlemmermeer.test.open-formulieren.nl/prefill-test-bsn
2. Use 900174341 or 900174316 for a BSN
3. Fill in e-mail and telephone number
4. Try to send the form

### Verwacht gedrag / Expected behavior

Form sends without warning and shows the confirmation page

### Device

Desktop

### OS

Windows

### Browser

Chrome, Firefox

**Comment by hlmr-erik** [comment]
Niemand een idee? Dit is best wel vervelend, nu kan ik het verzenden en de dankteksten/mails van DigiD-only formulieren niet testen tenzij ik ze vermink door de prefill uit te zetten.
![image](https://user-images.githubusercontent.com/111565013/190105187-9b8ebd72-20c5-4897-8e94-e8f979f14137.png)

**Comment by hlmr-erik** [comment]
The problem disappears when disabling 'read-only' on these fields (so the suggestion I made in the original ticket about applying normalisation is not true).

**Comment by sergei-maertens** [comment]
I suspect this is a formatting mismatch while validating that the prefill data has not been tampered with (which is only done with read-only fields).

**Comment by SilviaAmAm** [comment]
I think that this has indeed to do with the formatting. If the fields are 'read-only' the backend checks that the data returned by the prefill and the data that the user submits are the same.

In the case of the postcode and the date, i see that the data returned by Stuf-BG is `4505XY` and `19990101` respectively, while the submission data contains `4505 XY` and `1999-01-01`

**Comment by SilviaAmAm** [comment]
We had this issue: https://github.com/open-formulieren/open-forms/issues/1693 where a normalisation step was added before setting the values as default values of the components.

This normalisation step is only applied when setting the values (https://github.com/open-formulieren/open-forms/blob/master/src/openforms/prefill/__init__.py#L119), but not when caching them on the submission (https://github.com/open-formulieren/open-forms/blob/1.1.7/src/openforms/prefill/__init__.py#L109). The cached values are used for the check (https://github.com/open-formulieren/open-forms/blob/1.1.7/src/openforms/submissions/api/validation.py#L109).

**Comment by hlmr-erik** [comment]
This still occurs in 2.0.0, just with a different error message. Is this correct, eg. would it be fixed only in 2.0.1?
```

{
    "type": "https://haarlemmermeer.test.open-formulieren.nl/fouten/ValidationError/",
    "code": "invalid",
    "title": "Invalid input.",
    "status": 400,
    "detail": "",
    "instance": "urn:uuid:3685bf60-8c86-47d6-8e04-7eef2c3a9062",
    "invalidParams": [
        {
            "name": "data.geboortedatum",
            "code": "invalidPrefilledField",
            "reason": "Je mag de prefill-gegevens niet wijzingen."
        },
        {
            "name": "data.postcode",
            "code": "invalidPrefilledField",
            "reason": "Je mag de prefill-gegevens niet wijzingen."
        }
    ]
}
```

**Comment by hlmr-erik** [comment]
@SilviaAmAm @sergei-maertens
This still occurs in 2.0.0, just with a different error message. Is this correct, eg. would it be fixed only in 2.0.1?

```
{
    "type": "https://haarlemmermeer.test.open-formulieren.nl/fouten/ValidationError/",
    "code": "invalid",
    "title": "Invalid input.",
    "status": 400,
    "detail": "",
    "instance": "urn:uuid:3685bf60-8c86-47d6-8e04-7eef2c3a9062",
    "invalidParams": [
        {
            "name": "data.geboortedatum",
            "code": "invalidPrefilledField",
            "reason": "Je mag de prefill-gegevens niet wijzingen."
        },
        {
            "name": "data.postcode",
            "code": "invalidPrefilledField",
            "reason": "Je mag de prefill-gegevens niet wijzingen."
        }
    ]
}
```

**Comment by sergei-maertens** [comment]
no, this should have been fixed in 2.0.0 and 1.1.8 - this is a bit hard to reason about without seeing which data was retrieved from prefill and what was eventually submitted though. Can you send us that (privately, in case the data is privacy sensitive)?

**Comment by hlmr-erik** [comment]
Its all dummy data so it can be posted here :)
I've asked someone with access to the prefill what data it gave back. In the meantime, this is the information that is submitted:
{
	"data": {
		"achternaam": "Een grotere naamëáõłćĉċčḉŞ",
		"achternaam1": "",
		"bedrijfsnaam": "",
		"burgerservicenummerBsn": "900174328",
		"email": "devops_testteam@haarlemmermeer.nl",
		"eMailadres": "",
		"geboortedatum": "19730121",
		"huisnummer": "1",
		"huisnummerletter": "",
		"huisnummertoevoeging": "",
		"plaats": "Hoofddorp",
		"plaats1": "",
		"postcode": "2132TZ",
		"straat": "Raadhuisplein",
		"straat1": "",
		"telefoonnummer": "0645454555",
		"telefoonnummer1": "",
		"tussenvoegselS": "",
		"tussenvoegselS1": "",
		"vestiging": "",
		"voorletterS": "",
		"voornaam": "Voornaam"
	},
	"metadata": {
		"browserName": "Netscape",
		"offset": 60,
		"onLine": true,
		"origin": "https://haarlemmermeer.test.open-formulieren.nl",
		"pathName": "/aanvraag-toestemming-parkeren/stap/uw-gegevens-digid-eherk-contactpersoon",
		"referrer": "https://haarlemmermeer.test.open-formulieren.nl/auth/aanvraag-toestemming-parkeren/demo/start?next=https%3A%2F%2Fhaarlemmermeer.test.open-formulieren.nl%2Faanvraag-toestemming-parkeren%2F%3F_start%3D1",
		"timezone": "Europe/Amsterdam",
		"userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:106.0) Gecko/20100101 Firefox/106.0"
	},
	"state": "submitted"
}

**Comment by SilviaAmAm** [comment]
Mhmm I see that the geboortedatum and the postcode are not normalised (https://github.com/open-formulieren/open-forms/blob/master/src/openforms/prefill/__init__.py#L118) :thinking:

**Comment by sergei-maertens** [comment]
Back on the sprint board then :) I'll leave this to you @SilviaAmAm

**Comment by sergei-maertens** [comment]
Applied a number of patches to the stable/2.0.x branch too, so this will be included in 2.0.11.

For the stable/1.1.x branch this was indeed correctly fixed, however the prefill internal mechanism was changed for 2.0 to use variables and that caused us to overlook this (before 2.0, prefill data was not properly stored and fetched every time instead).
